SwordPlugin.inc.php

```diff
@@ -0,0 +1,304 @@
+
+		// Check the operation.
+		$op = $args[1];
+
+		if ($op == 'swordSettings') { // settings tab
+			define('HANDLER_CLASS', 'SwordSettingsTabHandler');
+			$args[2] = $this->getPluginPath() . '/' . 'SwordSettingsTabHandler.inc.php';
+		}
+		else {
+			$publicOps = array(
+				'depositPoints',
+				'performManagerOnlyDeposit',
+				'index',
+			);
+
+			if (!in_array($op, $publicOps)) return;
+
+			define('HANDLER_CLASS', 'SwordHandler');
+			$args[2] = $this->getPluginPath() . '/' . 'SwordHandler.inc.php';
+		}
+	}
+
+	/**
+	 * Extend the website settings tabs to include sword settings
+	 * @param $hookName string The name of the invoked hook
+	 * @param $args array Hook parameters
+	 * @return boolean Hook handling status
+	 */
+	public function callbackSettingsTab($hookName, $args) {
+		$output =& $args[2];
+		$request =& Registry::get('request');
+		$dispatcher = $request->getDispatcher();
+		$tabLabel = __('plugins.generic.sword.displayName') . ' ' . __('plugins.generic.sword.settings');
+		$output .= '<li><a name="swordSettings" id="swordSettings" href="' . $dispatcher->url($request, ROUTE_PAGE, null, 'sword', 'swordSettings') . '">' . $tabLabel . '</a></li>';
+		return false;
+	}
+
+	/**
+	 * Permit requests to SWORD deposit points grid handler
+	 * @param $hookName string The name of the hook being invoked
+	 * @param $args array The parameters to the invoked hook
+	 */
+	public function setupGridHandler($hookName, $params) {
+		$component = $params[0];
+		if ($component == 'plugins.generic.sword.controllers.grid.SwordDepositPointsGridHandler') {
+			import($component);
+			SwordDepositPointsGridHandler::setPlugin($this);
+			return true;
+		}
+		if ($component == 'plugins.generic.sword.controllers.grid.SubmissionListGridHandler') {
+			import($component);
+			SubmissionListGridHandler::setPlugin($this);
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Get the display name of this plugin
+	 * @return string
+	 */
+	public function getDisplayName() {
+		return __('plugins.generic.sword.displayName');
+	}
+
+	/**
+	 * Get the description of this plugin
+	 * @return string
+	 */
+	public function getDescription() {
+		return __('plugins.generic.sword.description');
+	}
+
+	/**
+	 * @see Plugin::getActions()
+	 */
+	public function getActions($request, $verb) {
+		$router = $request->getRouter();
+		$dispatcher = $request->getDispatcher();
+		import('lib.pkp.classes.linkAction.request.RedirectAction');
+		return array_merge(
+			// Settings
+			$this->getEnabled()?array(
+				new LinkAction(
+					'swordSettings',
+					new RedirectAction($dispatcher->url(
+						$request, ROUTE_PAGE,
+						null, 'management', 'settings', 'website',
+						array('uid' => uniqid()),
+						'swordSettings'
+						)),
+					__('manager.plugins.settings'),
+					null
+					),
+			):array(),
+			parent::getActions($request, $verb)
+		);
+	}
+
+	/**
+	 * @see Plugin::manage()
+	 */
+	public function manage($args, $request) {
```

Comment:
It doesn't look like this code does anything.

Reply:
removed.